src/python/pants/backend/docker/goals/package_image_test.py

```diff
@@ -1152,7 +1152,7 @@ def test_docker_cache_from_option(rule_runner: RuleRunner) -> None:
                 """\
                 docker_image(
                   name="img1",
-                  cache_from={"type": "local", "dest": "/tmp/docker/pants-test-cache"},
```

Comment:
Does this old value continue to work? Or does it become a hard error after this change?

Reply:
It will be a hard error as it is validating against a list in`target.py` below

Reply:
Ah I see. That's potentially something we should try to improve, given this is a feature released as stable and even highlighted in release notes (e.g. https://www.pantsbuild.org/blog/2024/01/24/pants-2-19#docker-improvements), so making a change like this would mean that any users who've adopted it will have a harder time upgrading.

Do we have any path for doing this with some sort of a deprecation and guidance for users? I can think of:

- add this as a field with a new name and deprecate the old one
- allow it to be either the `dict` or a `list` of such
- anything else?

Reply:
Wow, nice catch @huonw 

I agree that translating a dict value into a single element list wrapping the dict is a sensible thing to do.

Question is if we want to accept this as a feature long term, or if we want to issue a warning doing so, to urge users to explicitly use a single value list?

I think supporting this as a feature does align with how list options are supported on the cli. I.e. `--flag=1 --flag=2` collects into a list option of `[1, 2]`. It's not the same, for sure, but the semantics are close enough I think.